vms/platformvm/txs/executor/camino_tx_executor.go

```diff
@@ -780,8 +784,15 @@ func (e *CaminoStandardTxExecutor) RegisterNodeTx(tx *txs.RegisterNodeTx) error
 	newNodeIDEmpty := tx.NewNodeID == ids.EmptyNodeID
 	oldNodeIDEmpty := tx.OldNodeID == ids.EmptyNodeID
 
-	if oldNodeIDEmpty && !newNodeIDEmpty &&
-		consortiumMemberAddressState&txs.AddressStateRegisteredNodeBit != 0 {
+	linkedNodeID, err := e.State.GetShortLink(tx.ConsortiumMemberAddress, state.ShortLinkKeyRegisterNode)
+	haslinkedNode := true
+	if err == database.ErrNotFound {
+		haslinkedNode = true
```

Comment:
I think this should be `haslinkedNode = false` instead of true
But I'd replace `haslinkedNode := true` with `haslinkedNode := err != database.ErrNotFound` and leave only last if with `err != nil`